### PR TITLE
Sentry and Google Analytics

### PR DESCRIPTION
**Pull request by nein09**
I don't normally like to pile multiple fixes into one branch, but here we are. This has:

- an upgrade to the Sentry Python API
- Better handling for 405 errors, because people are using bots to try weird stuff on our server and it's clogging up our error reporting system with junk
- A way to set the Google Analytics IDs as an environment variable, so that we can set different IDs per-deployment, using Helm

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -36,13 +33,15 @@
 except OSError:
     pass
 
+
+@app.context_processor
```
Comment (nein09) [inline app/__init__.py:37]: This makes a variable, `google_analytics_id` available to the front-end templates

File: app/templates/base.html [hunk 2]
```diff
@@ -6,16 +6,17 @@
...
-     <!-- Global site tag (gtag.js) - Google Analytics -->
-    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-234464538-2"></script>
-    <script>
-      window.dataLayer = window.dataLayer || [];
-      function gtag(){dataLayer.push(arguments);}
-      gtag('js', new Date());
-
-      gtag('config', 'UA-234464538-2');
-    </script>
+     {% if google_analytics_id %}
```
Comment (nein09) [inline app/templates/base.html:9]: if we have a google analytics id, we can include it in the page

File: dev.env [hunk 3]
```diff
@@ -10,6 +10,7 @@ export FLASK_ENV=development
 
 # Setup for Sentry, our error tracking software
 export SENTRY_DSN=
+export SENTRY_ENVIRONMENT=development
```
Comment (nein09) [inline dev.env:13]: This has been in my `.env` for ages but I realized that I hadn't shared it with everyone.

File: dev.env [hunk 4]
```diff
@@ -29,3 +30,6 @@ export GLEANER_GRAPH=${GLEANER_BASE}/datavol/graph
 
 # Kubernetes deployment helper
 export KUBECONFIG=~/.kube/config:~/polder-federated-search/helm/polder.config
+
+# Google Analytics property ID
+export GOOGLE_ANALYTICS_ID=
```
Comment (nein09) [inline dev.env:35]: After this change gets merged, you'll need to add this line in your `.env` for Google events to show up from your development machine.

File: helm/values-dev.yaml [hunk 5]
```diff
@@ -116,6 +116,7 @@ GDB_JAVA_OPTS: >-
   -Dreuse.vars.in.subselects=true
 
 sentryEnvironment: development
+googleAnalyticsId: UA-234464538-2
```
Comment (nein09) [inline helm/values-dev.yaml:119]: A new GA Id for search-dev can go here if we want one

File: helm/values-local.yaml [hunk 6]
```diff
@@ -111,6 +111,7 @@ GDB_JAVA_OPTS: >-
   -Dreuse.vars.in.subselects=true
 
 sentryEnvironment: local
+googleAnalyticsId: UA-234464538-2
```
Comment (nein09) [inline helm/values-local.yaml:114]: A new GA ID for development can go here, if we want one

**Review by yemoski — APPROVED** [review]

**Pull request merged**